cmd/translatesfx/translatesfx/otel_test.go

```diff
@@ -318,6 +318,44 @@ func TestInfoToOtelConfig_MetricsToExclude_Monitor(t *testing.T) {
 	}, ex)
 }
 
+func TestComponentCollection_Single(t *testing.T) {
+	cc := componentCollection{{
+		provisionalKey: "mycomponent",
+		attrs:          map[string]interface{}{"foo": "bar"},
+	}}
+	componentMap := cc.toComponentMap()
+	var keys []string
+	for k := range componentMap {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	assert.Equal(t, []string{"mycomponent"}, keys)
+}
+
+func TestComponentCollection_Multiple(t *testing.T) {
+	cc := componentCollection{{
+		provisionalKey: "mycomponent",
+		attrs:          map[string]interface{}{"foo": "bar"},
+	}, {
+		provisionalKey: "mycomponent",
+		attrs:          map[string]interface{}{"foo": "bar"},
+	}}
+	componentMap := cc.toComponentMap()
+	var keys []string
+	for k := range componentMap {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	assert.Equal(t, []string{"mycomponent/0", "mycomponent/1"}, keys)
+}
+
+func TestInfoToOtelConfig_DuplicateMonitors(t *testing.T) {
+	cfg, _ := yamlToOtelConfig(t, "testdata/sa-duplicate-monitors.yaml")
+	assert.Equal(t, 2, len(cfg.Receivers))
+	metrics := cfg.Service.Pipelines["metrics"]
+	assert.Equal(t, []string{"smartagent/sql/0", "smartagent/sql/1"}, metrics.Receivers)
```

Comment:
Do we have any coverage that the different receiver attributes resolve as desired in the final config?

Reply:
I had only tested that manually -- added coverage.-